ethereum/contracts/bridge/Bridge.sol

```diff
@@ -410,14 +491,15 @@ contract Bridge is BridgeGovernance {
 
         transfer.to = encoded.toBytes32(index);
         index += 32;
-
+        
         transfer.toChain = encoded.toUint16(index);
         index += 2;
 
         transfer.fee = encoded.toUint256(index);
         index += 32;
 
-        require(encoded.length == index, "invalid Transfer");
+        // payload 3 allows for an arbitrary additional payload
+        require(encoded.length == index || transfer.payloadID == 3, "invalid Transfer");
```

Comment:
this doesn't set `transfer.payload` (which doesn't exist, because the function return `Transfer` and not `TransferWithPayload`)

Reply:
Just want to add to this: right now this function returns the whole VAA binary, and not just the payload (in case there is a payload). It's workable, clients just have to index into the binary at the right offset, but that essentially means leaking out the signature protocol, and changing it in the future would be a breaking change for downstream users of payload 3.